src/SDKs/IotCentral/IotCentral.Tests/ScenarioTests/IotCentralLifeCycleTests.cs

```diff
@@ -86,43 +87,53 @@ public void TestIotCentralUpdateLifeCycle()
                 // Create Resource Group
                 var resourceGroup = CreateResourceGroup(IotCentralTestUtilities.DefaultUpdateResourceGroupName);
 
-                // Check if App Exists and Delete
-                DeleteAppIfExists(IotCentralTestUtilities.DefaultUpdateResourceGroupName, IotCentralTestUtilities.DefaultUpdateIotCentralName);
-
                 // Create App
-                var app = CreateIotCentral(resourceGroup, IotCentralTestUtilities.DefaultLocation, IotCentralTestUtilities.DefaultUpdateIotCentralName);
+                var app = CreateIotCentral(resourceGroup, IotCentralTestUtilities.DefaultLocation, IotCentralTestUtilities.DefaultUpdateResourceName, IotCentralTestUtilities.DefaultUpdateSubdomain);
+
+                // Validate resourceName and subdomain
+                this.ValidateAppParameters(app.Name, app.Subdomain);
 
                 // Update App
                 var newSubDomain = "test-updated-sub-domain";
                 var newDisplayName = "test-updated-display-name";
+                // Add and Get Tags
+                IDictionary<string, string> tags = new Dictionary<string, string>
+                {
+                    { "key1", "value1" },
+                    { "key2", "value2" }
+                };
 
-                app.Subdomain = newSubDomain;
-                app.DisplayName = newDisplayName;
-                app = UpdateIotCentral(resourceGroup, app, IotCentralTestUtilities.DefaultUpdateIotCentralName);
+                AppPatch appPatch = new AppPatch()
+                {
+                    Tags = tags,
+                    DisplayName = newDisplayName,
+                    Subdomain = newSubDomain
+                };
+
+                app = UpdateIotCentral(resourceGroup, appPatch, IotCentralTestUtilities.DefaultUpdateResourceName);
 
                 // List apps
                 app = iotCentralClient.Apps.ListByResourceGroup(IotCentralTestUtilities.DefaultUpdateResourceGroupName)
-                    .FirstOrDefault(e => e.Name.Equals(IotCentralTestUtilities.DefaultUpdateIotCentralName, StringComparison.OrdinalIgnoreCase));
+                    .FirstOrDefault(e => e.Name.Equals(IotCentralTestUtilities.DefaultUpdateResourceName, StringComparison.OrdinalIgnoreCase));
 
                 Assert.NotNull(app);
                 Assert.Equal(newDisplayName, app.DisplayName);
-                Assert.Equal(newSubDomain, app.Subdomain);
+                Assert.True(app.Tags.Count().Equals(2));
+                Assert.Equal("value2", app.Tags["key2"]);
             }
         }
 
-        private void DeleteAppIfExists(string resourceGroupName, string name)
+        private void ValidateAppParameters(string resourceName, string subdomain)
```

Comment:
nit: change this method name to something that reflects that it checks the names to make sure they are marked as taken

Reply:
Changed